pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go

```diff
@@ -516,6 +523,12 @@ func (s *Container) getStatsForStmtWithKey(
 func (s *Container) getStatsForStmtWithKeyLocked(
 	key stmtKey, stmtFingerprintID appstatspb.StmtFingerprintID, createIfNonexistent bool,
 ) (stats *stmtStats, created, throttled bool) {
+	func() {
+		s.serverCardinality.Lock()
+		defer s.serverCardinality.Unlock()
+		s.serverCardinality.cardinality.Insert([]byte(key.stmtNoConstants))
```

Comment:
Sorry I'm late on this. Why are we basing it off of the representative query rather than the id itself? Is the goal here just to measure cardinality due to the representative sql query?